app/assets/javascripts/services/planService.js

```diff
@@ -144,6 +158,37 @@ planner.factory('planService', ['Restangular', '_', function(Restangular, _) {
     plan.completed_courses = _.filter(_.values(plan.coursesById), function(course) {
       return course.completed === true;
     });
+
+    if (plan.elective_courses) {
+      _replaceElectives(plan.elective_courses, plan.intended_courses, 'intended');
+      _replaceElectives(plan.elective_courses, plan.completed_courses, 'completed');
+    }
+  };
+
+  var _replaceElectives = function(elective_courses, courses, property) {
+    var count, course_index, elective_index;
+
+    // go through intended courses
+    for (var i = 0; i < courses.length; i++) {
```

Comment:
This function is ugly, but not refactored because it will need to be changed to accommodate if a single course is selected as an elective multiple times, like `thesis`, or if the elective is also a required course, like `directed practicum`.
I would skip this if I were you.